packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -170,6 +170,75 @@ export default class RoutesHandler {
         }
     };
 
+    /**
+     * Handler for retrieving existing controller extension data from the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetControllerExtensionData = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const params = req.params as { controllerName: string };
+            const controllerName = sanitize(params.controllerName);
+            const codeExtFiles = await this.readAllFilesByGlob('/**/changes/*_codeExt.change');
+
+            let controllerExists = false;
+            let controllerPath = '';
+            let controllerPathFromRoot = '';
+
+            const project = this.util.getProject();
+            const sourcePath = project.getSourcePath();
+            const rootPath = project.getRootPath();
+            const projectName = rootPath.split(/[\\/]/).slice(-1)[0];
+
+            const getPath = (projectPath: string, fileName: string) =>
+                path.join(projectPath, FolderNames.Changes, FolderNames.Coding, fileName).replace(/[\\/]/g, path.sep);
+
+            for (const file of codeExtFiles) {
+                const fileStr = await file.getString();
+                const change = JSON.parse(fileStr) as {
+                    selector: {
+                        controllerName: string;
+                    };
+                    content: {
+                        codeRef: string;
+                    };
+                };
+
+                if (change.selector.controllerName === controllerName) {
+                    const fileName = change.content.codeRef.replace('coding/', '');
+                    controllerPath = getPath(sourcePath, fileName);
+                    controllerPathFromRoot = getPath(projectName, fileName);
+                    controllerExists = true;
+                    break;
+                }
+            }
+
+            if (controllerExists && !fs.existsSync(controllerPath)) {
+                const errorMsg = `Controller extension file was not found at ${controllerPath}`;
+                res.status(HttpStatusCodes.NOT_FOUND).send({
+                    message: errorMsg
+                });
+                this.logger.debug(errorMsg);
+                return;
+            }
```

Comment:
Can we rewrite this in
```
if (change.selector.controllerName === controllerName) {
    const fileName = change.content.codeRef.replace('coding/', '');
    controllerPath = getPath(sourcePath, fileName);
   if(fs.existSync) {
    // Read file
} else {
//Throw an error
}

}
```

Reply:
Changed in: https://github.com/SAP/open-ux-tools/pull/1408/commits/5ecb566fd4fb6f2a6d62ff1bceecce59d914acc7